library/lib-core-logging/src/main/java/com/sokima/lib/corelogging/usecase/impl/MaskUseCaseImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.sokima.lib.corelogging.usecase.impl;
+
+import com.sokima.lib.corelogging.usecase.MaskUseCase;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+abstract class MaskUseCaseImpl implements MaskUseCase {
+    protected static final String MASK = "****";
+    private String fieldPattern;
+    private Pattern pattern;
+
+    protected MaskUseCaseImpl(List<String> fields, String tamplatePattern) {
+        this.fieldPattern = String.join("|", fields);
+        this.pattern = Pattern.compile(String.format(tamplatePattern, fieldPattern));
+    }
+
+    public String maskMessage(String message, String regex) {
+        StringBuffer buffer = new StringBuffer();
```

Comment:
Any sense of using StringBuffer here. use StringBuilder 
```suggestion
        StringBuffer buffer = new StringBuffer();
```